pandas/core/reshape/merge.py

```diff
@@ -886,23 +886,20 @@ def _get_join_info(self):
             if self.right_index:
                 if len(self.left) > 0:
                     join_index = self._create_join_index(
-                        self.left.index,
-                        self.right.index,
-                        left_indexer,
-                        right_indexer,
-                        how="right",
+                        self.left.index, self.right.index, left_indexer, how="right",
                     )
                 else:
                     join_index = self.right.index.take(right_indexer)
                     left_indexer = np.array([-1] * len(join_index))
             elif self.left_index:
-                if len(self.right) > 0:
+                if self.how == "asof":
                     join_index = self._create_join_index(
-                        self.right.index,
-                        self.left.index,
-                        right_indexer,
-                        left_indexer,
-                        how="left",
+                        self.left.index, self.right.index, left_indexer, how="left",
+                    )
+
+                elif len(self.right) > 0:
+                    join_index = self._create_join_index(
```

Comment:
shouldn't this be how='right'?

Reply:
We could change that, that would actually help quite a lot with other issues. But I would prefer doing that in another pull request, because that will break a lot of tests depending on the wrong behavior. Is this all right with you?

Reply:
ok great - yeah this looks wrong but let's address later (iirc u have another PR out about this)

Reply:
Yes, you are right. Tried a different approach there, but this should be way better. I will update that pr the coming days